qubesmanager/qube_manager.py

```diff
@@ -0,0 +1,1197 @@
+    settings_loaded = False
+    columns_indices = {"Type": 0,
+                       "Label": 1,
+                       "Name": 2,
+                       "State": 3,
+                       "Template": 4,
+                       "NetVM": 5,
+                       "Size": 6,
+                       "Internal": 7,
+                       "IP": 8,
+                       "Backups": 9,
+                       "Last backup": 10,
+                       }
+
+    def __init__(self, qubes_app, qt_app, parent=None):
+        # pylint: disable=unused-argument
+        super(VmManagerWindow, self).__init__()
+        self.setupUi(self)
+        self.toolbar = self.toolBar
+
+        self.manager_settings = QtCore.QSettings(self)
+
+        self.qubes_app = qubes_app
+        self.qt_app = qt_app
+
+        self.searchbox = SearchBox()
+        self.searchbox.setValidator(QtGui.QRegExpValidator(
+            QtCore.QRegExp("[a-zA-Z0-9-]*", QtCore.Qt.CaseInsensitive), None))
+        self.searchContainer.addWidget(self.searchbox)
+
+        self.connect(self.table, QtCore.SIGNAL("itemSelectionChanged()"),
+                     self.table_selection_changed)
+
+        self.table.setColumnWidth(0, self.column_width)
+
+        self.sort_by_column = "Type"
+        self.sort_order = QtCore.Qt.AscendingOrder
+
+        self.vms_list = []
+        self.vms_in_table = {}
+        self.reload_table = False
+
+        self.frame_width = 0
+        self.frame_height = 0
+
+        self.move(self.x(), 0)
+
+        self.columns_actions = {
+            self.columns_indices["Type"]: self.action_vm_type,
+            self.columns_indices["Label"]: self.action_label,
+            self.columns_indices["Name"]: self.action_name,
+            self.columns_indices["State"]: self.action_state,
+            self.columns_indices["Template"]: self.action_template,
+            self.columns_indices["NetVM"]: self.action_netvm,
+            self.columns_indices["Size"]: self.action_size_on_disk,
+            self.columns_indices["Internal"]: self.action_internal,
+            self.columns_indices["IP"]: self
+                .action_ip, self.columns_indices["Backups"]: self
+                .action_backups, self.columns_indices["Last backup"]: self
+            .action_last_backup
+        }
+
+        self.visible_columns_count = len(self.columns_indices)
+        self.table.setColumnHidden(self.columns_indices["Size"], True)
+        self.action_size_on_disk.setChecked(False)
+        self.table.setColumnHidden(self.columns_indices["Internal"], True)
+        self.action_internal.setChecked(False)
+        self.table.setColumnHidden(self.columns_indices["IP"], True)
+        self.action_ip.setChecked(False)
+        self.table.setColumnHidden(self.columns_indices["Backups"], True)
+        self.action_backups.setChecked(False)
+        self.table.setColumnHidden(self.columns_indices["Last backup"], True)
+        self.action_last_backup.setChecked(False)
```

Comment:
`load_manager_settings()` use just `self.action_*.setChecked()` to show/hide columns. `self.table.setColumnHidden()` looks unnecessary here.